batch_finder_resource_method.md

```diff
@@ -0,0 +1,314 @@
+Instead of callings multiple finders with different filter values, we call 1 BATCH_FINDER method with a list of filters.
+
+For example, a client might want to call the same FINDER with different search criteria.
+Combining multiple individual requests into a single batch request can save the application significant network latency.
+Also, the server can execute searches more efficiently if they are combined as a single query.
+
+BATCH FINDER should not have any visible side effects.
+For example, it should be safe to call whenever the client wishes.
+However, this is not something enforced by the framework, and it is up to the application developer that there are no side effects.
+
+It is important to note that:
+- the operations may execute on the server out of order
+- the response objects are expected to be returned in the same order and position as the respective input search criteria.
+- The BATCH_FINDER will require implementing a resource method to handle a BATCH_FINDER requests. It won't behave like a multiplexer that will call automatically existing finders
+
+## Protocol
+### Request
+See more details here [BatchFinderUri](/rest.li/spec/protocol#batch-finders).
+
+### Response
+See more details here [BatchCollectionResponse](/rest.li/spec/protocol#batch-collection-response).
+    
+    
+## Client
+### Java Request Builders
+The client framework includes a code-generation tool that reads the IDL(see [Restspec IDL](/rest.li/user_guide/restli_client#restspec-idl) for details) and generates type-safe Java binding for each resource and its supported methods.
+The bindings are represented as RequestBuilder classes.  
+For each resource described in an IDL file, a corresponding builder factory will be generated.
+The factory contains a factory method for each resource method supported by the resource. 
+The factory method returns a request builder object with type-safe bindings for the given method.
+More details in [Resource Builder Factory](/rest.li/user_guide/restli_client#resource-builder-factory).
+
+For example:
+```java
+// Request builders factory class
+// which provides all the specific request builders to corresponding resource method in defined resource class
+public class GreetingsRequestBuilders extends BuilderBase {
+    public GreetingsRequestBuilders()
+    public GreetingsRequestBuilders(String primaryResourceName)
+    public GreetingsCreateRequestBuilder create()
+    public GreetingsGetRequestBuilder get()
+    ...
+    public GreetingsFindBySearchRequestBuilder findBySearch()
+    // the BATCH_FINDER resource method named "SearchGreetings"
+    public GreetingsBatchFindBySearchGreetingsRequestBuilder batchFindBySearchGreetings() 
+}
+```
+The request builders factory class provides a method to generate the corresponding BATCH_FINDER request builder.
+More details about generated class and method declaration in [BATCH FINDER Request Builder](/rest.li/user_guide/restli_client#batch-finder-request-builder).
```

Comment:
This document should provide all the details. The user guide should provide a simple use case. Having a link to the user-guide from this doc means the details should me moved here.